src/com/lftechnology/java/training/niraj/ems/services/implementations/UserServiceImpl.java

```diff
@@ -0,0 +1,507 @@
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * @param username
+     * @param password
+     * @return
+     * @throws SQLException
+     */
+    public Employee login(String username, String password) throws SQLException {
+        Map<String, String> credentials = new LinkedHashMap<String, String>();
+        credentials.put("username", username);
+        credentials.put("password", password);
+        credentials.put("status", Status.ACTIVE.getStatus());
+        credentials.put("isTerminated", Status.INACTIVE.getStatus());
+        Employee employee = employeeDao.findByAttributes(credentials);
+
+        return employee;
+    }
+
+    /**
+     * Saves user information
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * @param userInfo
+     * @return
+     * @throws SQLException
+     */
+    public String saveUserInfo(Map<String, String> userInfo) throws SQLException {
+        Employee employee = new Employee();
+        parseUserInfo(employee.getAttributes(), userInfo);
+        employeeDao.setEmployeeObject(userInfo, employee);
+        save(employee);
+        return employee.getId() != null ? employee.getId() : null;
+
+    }
+
+    /**
+     * Terminates an employee
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * @param id
+     * @return
+     * @throws SQLException
+     * @throws CustomException
+     */
+    public boolean terminateEmployee(String id) throws SQLException, CustomException {
+        Employee employee = employeeDao.find(id);
+        boolean isTerminated = false;
+        if (employee != null) {
+            employee.setIsTerminated(Status.ACTIVE);
+            isTerminated = employeeDao.update(employee);
+        } else {
+            throw new CustomException("Employee not found");
+        }
+        return isTerminated;
+    }
+
+    /**
+     * Deletes user
+     * 
+     * @param id
+     * @return
+     * @throws SQLException
+     * @throws CustomException
+     */
+    public boolean delete(String id) throws SQLException, CustomException {
+        Employee employee = employeeDao.find(id);
+        boolean isDeleted = false;
+        if (employee != null) {
+            employee.setStatus(Status.INACTIVE);
+            isDeleted = employeeDao.update(employee);
+        } else {
+            throw new CustomException("Employee not found");
+        }
+        return isDeleted;
+    }
+
+    /**
+     * Gets user registration informations
+     * 
+     * @author Niraj Rajbhandari<nirajrajbhandari@lftechnology.com>
+     * 
+     * @param scanner
+     * @param console
+     * @return
+     */
+    public Map<String, String> getUserRegistrationInfo(Scanner scanner, Console console) {
+        Map<String, String> userRegistrationInfo = new LinkedHashMap<String, String>();
+        userRegistrationInfo.put("fullname", userInput.getInput(scanner, console, Constants.FULLNAME_LABEL));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "fullname" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)